storage/aws/aws.go

```diff
@@ -183,15 +190,16 @@ func (s *Storage) ReadEntryBundle(ctx context.Context, i, sz uint64) ([]byte, er
 //
 // This is indended to be used to proxy read requests through the personality for debug/testing purposes.
 func (s *Storage) get(ctx context.Context, path string) ([]byte, error) {
-	d, _, err := s.objStore.getObject(ctx, path)
+	d, err := s.objStore.getObject(ctx, path)
 	return d, err
 }
 
 // init ensures that the storage represents a log in a valid state.
 func (s *Storage) init(ctx context.Context) error {
 	_, err := s.get(ctx, layout.CheckpointPath)
 	if err != nil {
-		if errors.Is(err, gcs.ErrObjectNotExist) {
+		var nske *types.NoSuchKey
+		if errors.As(err, &nske) {
```

Comment:
Since `nske` isn't used below, could you use `errors.Is` here?

Reply:
I thought as well, I did this first, and then ran into unexpected errors. Sadly,  I can't reproduce them.

The tl;dr is that `types.NoSuchKey` is a type, it's not an `error` returned with `errors.New()`, as opposed to `gcs. ErrObjectNotExist`. `errors.As()` does a type comparison, whereas `errors.Is()` does value comparison: see https://go.dev/blog/go1.13-errors.

I left a comment around to explain the rationale and make sure nobody undoes this.